tools/codegen/gen.py

```diff
@@ -171,14 +171,40 @@ def static_dispatch_keys(backend: Optional[BackendIndex]) -> List[DispatchKey]:
             DispatchKey.CompositeExplicitAutograd
         ]
 
-def static_dispatch_extra_headers(backend: Optional[BackendIndex], skip_tensor_include: bool = False) -> str:
+def get_static_dispatch_backend(f: NativeFunction, backend_index: BackendIndex) -> Optional[DispatchKey]:
+    if (f.structured_delegate is not None or backend_index.has_kernel(f)):
+        # TODO: for ops with structured_delegate it should check the dispatch table of
+        # the out variant instead. For now, these structured ops all have CPU/CUDA kernels
+        # so we always dispatch to the `backend`, but this could be wrong when we
+        # migrate math/default_backend ops to use structured delegate.
```

Comment:
cc @bdhirsh on this one.

Reply:
In case it isn't clear, this `TODO` comes from the existing function `static_dispatch` below. I've just factored it out into a separate function so I can use it to include the appropriate static dispatch header.

Reply:
Yes this is just a move. But Brian is working on this part of the codegen. So wanted to flag this for him in case he never saw this comment before.

Reply:
Sounds good. Mobile is working on some changes to static dispatch soon so we might have some merge conflicts. I'll let them know.